runtime/src/xor_fee_impls.rs

```diff
@@ -425,17 +433,55 @@ impl xor_fee::WithdrawFee<Runtime> for WithdrawFee {
         fee_source: &AccountId,
         call: &RuntimeCall,
         fee: Balance,
-    ) -> Result<(AccountId, Option<NegativeImbalanceOf<Runtime>>), DispatchError> {
+    ) -> Result<
+        (
+            AccountId,
+            Option<NegativeImbalanceOf<Runtime>>,
+            Option<AssetId>,
+        ),
+        DispatchError,
+    > {
         match call {
+            // TODO: remake for xorless
             RuntimeCall::Referrals(referrals::Call::set_referrer { referrer })
                 // Fee source should be set to referrer by `get_fee_source` method, if not 
                 // it means that user can't set referrer
                 if Referrals::can_set_referrer(who) =>
             {
                 Referrals::withdraw_fee(referrer, fee)?;
             }
+            #[allow(unused_variables)] // Xorless fee
+            RuntimeCall::XorFee(xor_fee::Call::xorless_call {call: _, asset_id}) => {
+                #[cfg(feature = "wip")] // Xorless fee
+                match *asset_id {
+                    None => {},
+                    Some(asset_id) if XorFee::whitelist_tokens().contains(&asset_id) => {
+                        let asset_fee = FixedWrapper::from(
+                            PriceTools::get_average_price(
+                                &GetXorAssetId::get(),
+                                &asset_id,
+                                PriceVariant::Buy)?
+                        ) * fee;
+                        let asset_fee = asset_fee.into_balance();
+                        if asset_fee.lt(&MinimalFeeInAsset::get()) {
```

Comment:
@vovac12 MinimalFeeInAsset used there as minimal amount for fee suitable for calculations.